src/backend/joanie/core/exceptions.py

```diff
@@ -29,3 +31,10 @@ class CertificateGenerationError(Exception):
     Exception raised when the certificate generation process fails due to the order not meeting
     all specified conditions.
     """
+
+
+class BackendTimeOut(ReadTimeout):
```

Comment:
```suggestion
class BackendTimeout(ReadTimeout):
```

Also, maybe you could instead raise that exception from the `delete_signing_procedure` directly, when a `ReadTimeout` is raised. I find it odd to subclass the `ReadTimeout` exception. WDYT?